packages/gatsby/src/commands/build.ts

```diff
@@ -310,6 +315,8 @@ module.exports = async function build(
       parentSpan: buildSpan,
       store,
     })
+
+    waitMaterializePageMode = materializePageMode()
```

Comment:
Just curious on why we're materializing after page queries here and in PQR mode we're doing it before.
If both can be moved to before or after we can call it once instead of twice.

Reply:
We don't really use non PQR mode, so I added it there "just in case" - as far as ordering, it generally doesn't matter - for PQR I start before query running due to main process just being largely idle and materialization can just overlap (time-wise) with running queries in workers

Reply:
Can't we do the following?
```
const waitMaterializePageMode = materializePageMode()
```

Reply:
I will move this, so those code path "match" more closely (even tho in practice it should be no-op as we never should use non-pqr code path these days)

Reply:
Ok, moved this outside of PQR if/else so we start it just before query running (and we still await on it before emitting ENGINGES_READY)